ai_helpdesk_agent/models/helpdesk_ticket.py

```diff
@@ -21,148 +18,165 @@ def send_ai_response(ticket_id, ai_result, user_id):
-                escalate = request_data.get('actions', [])
-                reasoning = request_data.get('reasoning', '') # TODO: where to use it?
-                self.save_ticket(ticket_id, escalate, continue_conv)
-                if text:
-                    ai_user_id = self.env['res.users'].search([('name', '=', 'AI Agent')], limit=1)
-                    send_ai_response(ticket_id, text, ai_user_id)
-            else:
-                if continue_conv:
-                    self.set_error_tag(ticket_id, 'AI Error', continue_conv)
-                    _logger.error(f'{ticket_id.id} AI Error, text: {request.text}, status: {request.status_code}')
-                else:
-                    send_default_email(ticket_id)
-            return
-        else:
-            self.set_error_tag(ticket_id, 'AI Error', continue_conv)
+        if dry_run and continue_conv or request.status_code != 200:
+            self.set_error_tag(ticket_id)
             _logger.error(f'{ticket_id.id} AI Error, text: {request.text}, status: {request.status_code}')
-
-    def set_error_tag(self, ticket_id, tag_name, continue_conv=False):
-        tag_id = self.env['helpdesk.tag'].search([('name', '=', tag_name)])
-        if continue_conv:
-            self.change_user(ticket_id, after_error=True)
-        else:
-            send_default_email(ticket_id)
+            return
+        request_data = request.json()
+        text = request_data.get('text', '')
+        escalate = request_data.get('actions', [])
+        reasoning = request_data.get('reasoning', '')  # TODO: where to use it?
+        self.save_ticket(ticket_id, escalate, continue_conv)
+        ai_user_id = get_ai_user(self.env)
+        if text:
+            send_ai_response(ticket_id, text, ai_user_id)
+        if reasoning:
+            ticket_id.message_post(body=reasoning, message_type='comment', subtype_xmlid='mail.mt_note')
+
+    @api.model
+    def set_error_tag(self, ticket_id, tag_name):
+        err_tag_id = self.env.ref('ai_helpdesk_agent.tag_ai_error')
         ticket_id.write({
-            'tag_ids': [Command.link(tag_id.id)],
+            'tag_ids': [Command.link(err_tag_id.id)],
         })
+        self.change_user(ticket_id, after_error=True)
+        send_default_email(ticket_id)
 
+    @api.model
     def send_request(self, data):
         api_key = self.env['ir.config_parameter'].sudo().get_param('ai_helpdesk_agent.api_key', '')
-        api_ulr = self.env['ir.config_parameter'].sudo().get_param('ai_helpdesk_agent.api_ulr')
+        api_ulr = self.env['ir.config_parameter'].sudo().get_param('ai_helpdesk_agent.api_ulr', '')
         headers = {
             'Content-Type': 'application/json',
             'X-API-KEY': api_key,
         }
         request = requests.post(api_ulr, json=data, headers=headers, timeout=20)
         return request
 
+    @api.model
     def save_ticket(self, ticket_id, escalate, continue_conv):
         try:
+            if 'SKIP' in escalate:
```

Comment:
Use named constants for SKIP and ESCALATE with explanation where they come from (or self-explaining names).